eppoclient/client_test.go

```diff
@@ -26,76 +24,64 @@ func Test_AssignBlankExperiment(t *testing.T) {
 
 	assert.Panics(t, func() {
-		_, err := client.GetStringAssignment("", "experiment-1", dictionary{})
+		_, err := client.GetStringAssignment("", "experiment-1", SubjectAttributes{}, "")
 		if err != nil {
 			log.Println(err)
 		}
 	})
 }
-
-func Test_SubjectNotInSample(t *testing.T) {
-	var mockLogger = new(mockLogger)
-	var mockConfigRequestor = new(mockConfigRequestor)
-	overrides := make(map[string]Value)
-	var mockVariations = []Variation{
-		{Name: "control", Value: String("control"), ShardRange: shardRange{Start: 0, End: 10000}},
-	}
-	var allocations = make(map[string]Allocation)
-	allocations[defaultAllocationKey] = Allocation{
-		PercentExposure: 0,
-		Variations:      mockVariations,
-	}
-	mockResult := experimentConfiguration{
-		Name:          "recommendation_algo",
-		Enabled:       true,
-		SubjectShards: 1000,
-		Overrides:     overrides,
-		Allocations:   allocations,
-		Rules:         []rule{defaultRule},
-	}
-
-	mockConfigRequestor.Mock.On("GetConfiguration", mock.Anything).Return(mockResult, nil)
-
-	client := newEppoClient(mockConfigRequestor, mockLogger)
-
-	assignment, err := client.GetStringAssignment("user-1", "experiment-key-1", dictionary{})
-
-	assert.Equal(t, "", assignment)
-	assert.NotNil(t, err)
-}
-
 func Test_LogAssignment(t *testing.T) {
 	var mockLogger = new(mockLogger)
 	mockLogger.Mock.On("LogAssignment", mock.Anything).Return()
 
 	var mockConfigRequestor = new(mockConfigRequestor)
-	overrides := make(map[string]Value)
-
-	var mockVariations = []Variation{
-		{Name: "control", Value: String("control"), ShardRange: shardRange{Start: 0, End: 10000}},
-	}
-	var allocations = make(map[string]Allocation)
-	allocations[defaultAllocationKey] = Allocation{
-		PercentExposure: 1,
-		Variations:      mockVariations,
-	}
-	mockResult := experimentConfiguration{
-		Name:          "recommendation_algo",
-		Enabled:       true,
-		SubjectShards: 1000,
-		Overrides:     overrides,
-		Allocations:   allocations,
-		Rules:         []rule{defaultRule},
-	}
-	mockConfigRequestor.Mock.On("GetConfiguration", "experiment-key-1").Return(mockResult, nil)
-
-	client := newEppoClient(mockConfigRequestor, mockLogger)
-
-	assignment, err := client.GetStringAssignment("user-1", "experiment-key-1", dictionary{})
+	var poller = newPoller(10, mockConfigRequestor.FetchAndStoreConfigurations)
+
+	config := map[string]flagConfiguration{
+		"experiment-key-1": flagConfiguration{
+			Key:           "experiment-key-1",
+			Enabled:       true,
+			VariationType: stringVariation,
+			Variations: map[string]variation{
+				"control": variation{
+					Key:   "control",
+					Value: "control",
+				},
+			},
+			Allocations: []allocation{
+				{
+					Key: "allocation-key",
+					Splits: []split{
+						{
+							VariationKey: "control",
+							Shards: []shard{
+								{
+									Salt: "",
+									Ranges: []shardRange{
+										{
+											Start: 0,
+											End:   10001,
```

Comment:
Shouldn't end be `10_000` instead of `10_001`? That is, the final bucket is 9999